src/lib.rs

```diff
@@ -75,16 +76,24 @@ pub enum HostType {
 
 impl From<c_uint> for HostType {
     fn from(value: c_uint) -> Self {
-        match value {
-            0 => Self::Domain,
-            1 => Self::IPV4,
-            2 => Self::IPV6,
-            _ => Self::Domain,
+        if value > 2 {
+            return Self::Domain;
         }
+
+        // Safety: Prior to transmuting we checked if value is bigger than 2
+        // and returned the default in that case.
+        //
+        // Mapping:
+        // - 0 => Domain
+        // - 1 => IPV4
+        // - 2 => IPV6
+        // - any other number maps to NotSpecial
+        unsafe { std::mem::transmute(value as u32) }
```

Comment:
```suggestion
        match value {
            0 => Self::Domain,
            1 => Self::IPV4,
            2 => Self::IPV6,
            _ => Self::Domain,
        }
```
If you replace the transmute with the same `match` block as before, the generated code (and therefore the performance) will be identical: https://x.com/tenellous/status/1819528511201464821

I'm all for performance improvement, but here there's only `unsafe` risk without any benefit.